packages/backend/src/apiKey/service.ts

```diff
@@ -0,0 +1,114 @@
+import { BaseService } from '../shared/baseService'
+import { SessionService } from '../session/service'
+import { ApiKey } from './model'
+import { uuid } from '../connector/core'
+import bcrypt from 'bcrypt'
+import { Transaction } from 'knex'
+import { SessionKey } from '../session/util'
+import { ApiKeyError } from './errors'
+
+export interface ApiKeyService {
+  create(apiKey: ApiKeyOptions, trx?: Transaction): Promise<NewApiKey>
+  get(apiKey: ApiKeyOptions): Promise<ApiKey[]>
+  redeem(sessionKey: SessionKeyOptions): Promise<SessionKey | ApiKeyError>
+  deleteAll(apiKey: ApiKeyOptions, trx?: Transaction): Promise<void>
+}
+
+interface ServiceDependencies extends BaseService {
+  sessionService: SessionService
+}
+
+type ApiKeyOptions = {
+  accountId: string
+}
+
+type SessionKeyOptions = {
+  accountId: string
+  key: string
+}
+
+interface NewApiKey extends ApiKey {
+  key: string
+}
+
+export async function createApiKeyService({
+  logger,
+  knex,
+  sessionService: sessionKeyService
+}: ServiceDependencies): Promise<ApiKeyService> {
+  const log = logger.child({
+    service: 'ApiKeyService'
+  })
+  const deps: ServiceDependencies = {
+    logger: log,
+    knex,
+    sessionService: sessionKeyService
+  }
+  return {
+    create: (options, trx) => createApiKey(deps, options, trx),
+    get: (options) => getApiKeys(deps, options),
+    redeem: (options) => redeemSessionKey(deps, options),
+    deleteAll: (options, trx) => deleteAllApiKeys(deps, options, trx)
+  }
+}
+
+async function createApiKey(
+  deps: ServiceDependencies,
+  { accountId }: ApiKeyOptions,
+  trx?: Transaction
+): Promise<NewApiKey> {
+  const keyTrx = trx || (await ApiKey.startTransaction(deps.knex))
```

Comment:
Either the commit and rollback should only happen `if (!trx)` (see [invoice creation](https://github.com/interledger/rafiki/blob/main/packages/backend/src/open_payments/invoice/service.ts#L82-L89)) or `createApiKey` could be transaction free (`ApiKey.query(deps.knex).insertAndFetch`).
I'd vote for the latter.

Reply:
I chatted to @sharafian about that at a point and he said I should make it consistent with what we do in other services. But I forgot to go back and check if all services to that. I checked now, they don't. And I agree, it is not really necessary to pass `trx` because I don't see a scenario where we need to create API keys atomically.